openedx/adg/admin_dashboard/admin_settings/managers.py

```diff
@@ -17,3 +17,11 @@ def get_invitation_for_email(self, email):
 
         statuses = (Invitation.PENDING, Invitation.ACCEPTED)
         return Invitation.objects.filter(email=email, status__in=statuses).first()
+
+    def get_cancelled_invitation_for_email(self, email):
+        """
+        Return invitation against an email if the status of the invitation is Cancelled
+        """
+        from openedx.adg.admin_dashboard.admin_settings.models import Invitation
+
+        return Invitation.objects.filter(email=email, status=Invitation.CANCELLED).last()
```

Comment:
Lets avoid this circular dependency by following these steps:
1. Move the choices from model to constants.
2. Import choices from constants in models and managers. 
3. Instead of using model name to filter, We can use self.objects.filter(email=email, status=CANCELLED).last()
4. This way circular dependency will be removed, Lets please start following this from now onwards.

Reply:
You will have to make changes in the method above this one as well.